whisperv6/whisper.go

```diff
@@ -760,17 +768,33 @@ func (whisper *Whisper) HandlePeer(peer *p2p.Peer, rw p2p.MsgReadWriter) error {
 		delete(whisper.peers, whisperPeer)
 		whisper.peerMu.Unlock()
 	}()
+	if whisper.ratelimiter != nil {
+		if err := whisper.ratelimiter.I.Create(whisperPeer.peer, whisper.ratelimiter.Config); err != nil {
+			return err
+		}
+		defer whisper.ratelimiter.I.Remove(whisperPeer.peer, 0)
+		defer whisper.ratelimiter.E.Remove(whisperPeer.peer, 0)
```

Comment:
By reading the code it's not clear why only `I` is created, but both `I` and `E` are deferred with Remove?
I found that `E` is created in `handshake()` later, but it looks like that code can be moved from `handshake()` to `HandlePeer()`, so all rate limiting logic is in one place. Does it make sense?

Reply:
it is created in the handhshake based on the info provided by the peer.